src/api/app/models/user.rb

```diff
@@ -261,27 +268,24 @@ def self.find_with_credentials(login, password)
       logger.debug( "No user found in database, creating" )
       logger.debug( "Email: #{ldap_info[0]}" )
       logger.debug( "Name : #{ldap_info[1]}" )
-      # Generate and store a fake pw in the OBS DB that no-one knows
-      chars = ["A".."Z", "a".."z", "0".."9"].collect { |r| r.to_a }.join
-      password = (1..24).collect { chars[rand(chars.size)] }.pack('a'*24)
+      # Generate and store a 24 char fake pw in the OBS DB that no-one knows
+      password = SecureRandom.base64
```

Comment:
Maybe a bit out-of-scope here, but the better approach would be not to set a random password, but some password which can not be used in real life. eg. we set it to "nil" and check for that on authentification in non-ldap case.
That would really eliminate any chance that someone can login if the LDAP config got disabled by accident.